### PR TITLE
Write a documentation for undocumented QACTL modules 

### PR DESCRIPTION
**Pull request by fernandolojano**
|Related issue|
|---|
| #1798  |

## Description
It is necessary to document all the existing classes and methods implemented in the PACTL modules. In order to do so, all the implemented modules had been reviewed and checked if they are documented.

This PR makes the following changes:
- Adds Documentation to the classes and methods that need it.

- [x] All the modules and its correspondent classes and functions are documented now

## Tasks 
The tasks made for this process where:
- [x] Search for undocumented modules from QACTL.
- [x] Write a documentation for all the modules that were found undocumented.

### CODE REVIEW

File: deps/wazuh_testing/wazuh_testing/qa_ctl/deployment/vagrantfile.py [hunk 1]
```diff
@@ -56,6 +56,12 @@ def __init__(self, file_path, box_image, vm_label, vm_name, cpus, memory, system
         self.box_url = self.__get_box_url()
 
     def __get_box_url(self):
+        """Gets the box URL of the specified box_image parameter
```
Comment (jmv74211) [inline deps/wazuh_testing/wazuh_testing/qa_ctl/deployment/vagrantfile.py:59]:
```suggestion
        """Get the box URL of the specified box_image parameter
```

File: deps/wazuh_testing/wazuh_testing/qa_ctl/provisioning/ansible/ansible_inventory.py [hunk 2]
```diff
@@ -80,6 +80,8 @@ def write_inventory_to_file(self):
             inventory.write(self.__str__())
 
     def delete_playbook_file(self):
+        """Delete all created playbook files
+        """
```
Comment (jmv74211) [inline deps/wazuh_testing/wazuh_testing/qa_ctl/provisioning/ansible/ansible_inventory.py:84]:
```suggestion
        """Delete all created playbook files"""
```

File: deps/wazuh_testing/wazuh_testing/qa_ctl/provisioning/qa_provisioning.py [hunk 3]
```diff
@@ -199,5 +199,7 @@ def run(self):
         QAProvisioning.LOGGER.info(f"The instances have been provisioned sucessfully")
 
     def destroy(self):
+        """Destroy all the temporary files created by an instance of this object
+        """
```
Comment (jmv74211) [inline deps/wazuh_testing/wazuh_testing/qa_ctl/provisioning/qa_provisioning.py:203]:
```suggestion
        """Destroy all the temporary files created by an instance of this object."""
```

File: deps/wazuh_testing/wazuh_testing/qa_ctl/provisioning/wazuh_deployment/wazuh_installation.py [hunk 4]
```diff
@@ -5,6 +5,18 @@
...
 
 class WazuhInstallation(ABC):
+    """Install a Wazuh instance set from the configuration file
+
+    Args:
+        wazuh_target (string): Type of the wazuh installation desired (manager or agent).
+        installation_files_path (string): Path where the Wazuh instalation files are located.
+        qa_ctl_configuration (QACTLConfiguration): QACTL configuration.
+    
+    Aributes:
```
Comment (jmv74211) [inline deps/wazuh_testing/wazuh_testing/qa_ctl/provisioning/wazuh_deployment/wazuh_installation.py:15]:
```suggestion
    Attributes:
```

File: deps/wazuh_testing/wazuh_testing/qa_ctl/provisioning/wazuh_deployment/wazuh_installation.py [hunk 5]
```diff
@@ -13,6 +25,13 @@ def __init__(self, wazuh_target, installation_files_path, qa_ctl_configuration):
 
     @abstractmethod
     def download_installation_files(self, inventory_file_path, ansible_tasks, hosts='all'):
+        """Downloads the installation files of Wazuh  by creating an ansible playbook and launching it
```
Comment (jmv74211) [inline deps/wazuh_testing/wazuh_testing/qa_ctl/provisioning/wazuh_deployment/wazuh_installation.py:28]:
```suggestion
        """Download the installation files of Wazuh  by creating an ansible playbook and launching it
```

File: deps/wazuh_testing/wazuh_testing/qa_ctl/provisioning/wazuh_deployment/wazuh_local_package.py [hunk 6]
```diff
@@ -9,6 +9,24 @@
...
+        system (string): System of the Wazuh installation files. Parameter set by default to 'None'.
+
+    Attributes:
+        local_package_path (string): Path where the local package is located.
+        package_name (string): name of the Wazuh package.
+        installation_files_path (string): Path where the Wazuh installation files are located.
+        local_package_path (string): Path where the local package is located.
+        qa_ctl_configuration (QACTLConfiguration): QACTL configuration.
+
+    """
```
Comment (jmv74211) [inline deps/wazuh_testing/wazuh_testing/qa_ctl/provisioning/wazuh_deployment/wazuh_local_package.py:29]:
```suggestion
    """
```

File: deps/wazuh_testing/wazuh_testing/qa_ctl/provisioning/wazuh_deployment/wazuh_s3_package.py [hunk 7]
```diff
@@ -9,6 +9,34 @@
 
 
 class WazuhS3Package(WazuhPackage):
+    """Install Wazuh from a S3 URL package
+
+
```
Comment (jmv74211) [inline deps/wazuh_testing/wazuh_testing/qa_ctl/provisioning/wazuh_deployment/wazuh_s3_package.py:14]:
```suggestion

```

File: deps/wazuh_testing/wazuh_testing/qa_ctl/provisioning/wazuh_deployment/wazuh_s3_package.py [hunk 8]
```diff
@@ -9,6 +9,34 @@
...
+        s3_package_url (string): URL of the S3 Wazuh package.
+        installation_files_path (string): Path where is located the Wazuh instalation files.
+        qa_ctl_configuration (QACTLConfiguration): QACTL configuration.
+        version (string): The version of Wazuh. Parameter set by default to 'None'.
+        system (string): System of the Wazuh installation files. Parameter set by default to 'None'.
+        revision (string): Revision of the wazuh package. Parameter set by default to 'None'.
+        repository (string): Repository of the wazuh package. Parameter set by default to 'None'.
+        architecture (string): Architecture of the Wazuh package. Parameter set by default to 'None'.
+
+
```
Comment (jmv74211) [inline deps/wazuh_testing/wazuh_testing/qa_ctl/provisioning/wazuh_deployment/wazuh_s3_package.py:26]:
```suggestion

```

**Review by jmv74211 — CHANGES_REQUESTED** [review]

File: deps/wazuh_testing/wazuh_testing/qa_ctl/deployment/vagrantfile.py (hunk 1 above)
Comment (fernandolojano) [reply deps/wazuh_testing/wazuh_testing/qa_ctl/deployment/vagrantfile.py]: The new modification can be reviewed in the commit: [bba01140837453f908a9dafefa89308ee9a5103e](https://github.com/wazuh/wazuh-qa/commit/bba01140837453f908a9dafefa89308ee9a5103e)

File: deps/wazuh_testing/wazuh_testing/qa_ctl/provisioning/ansible/ansible_inventory.py (hunk 2 above)
Comment (fernandolojano) [reply deps/wazuh_testing/wazuh_testing/qa_ctl/provisioning/ansible/ansible_inventory.py]: The new modification can be reviewed in the commit:[bba01140837453f908a9dafefa89308ee9a5103e](https://github.com/wazuh/wazuh-qa/commit/bba01140837453f908a9dafefa89308ee9a5103e)

File: deps/wazuh_testing/wazuh_testing/qa_ctl/provisioning/qa_provisioning.py (hunk 3 above)
Comment (fernandolojano) [reply deps/wazuh_testing/wazuh_testing/qa_ctl/provisioning/qa_provisioning.py]: The new modification can be reviewed in the commit:[bba01140837453f908a9dafefa89308ee9a5103e](https://github.com/wazuh/wazuh-qa/commit/bba01140837453f908a9dafefa89308ee9a5103e)

File: deps/wazuh_testing/wazuh_testing/qa_ctl/provisioning/wazuh_deployment/wazuh_installation.py (hunk 4 above)
Comment (fernandolojano) [reply deps/wazuh_testing/wazuh_testing/qa_ctl/provisioning/wazuh_deployment/wazuh_installation.py]: The new modification can be reviewed in the commit:[bba01140837453f908a9dafefa89308ee9a5103e](https://github.com/wazuh/wazuh-qa/commit/bba01140837453f908a9dafefa89308ee9a5103e)

File: deps/wazuh_testing/wazuh_testing/qa_ctl/provisioning/wazuh_deployment/wazuh_installation.py (hunk 5 above)
Comment (fernandolojano) [reply deps/wazuh_testing/wazuh_testing/qa_ctl/provisioning/wazuh_deployment/wazuh_installation.py]: The new modification can be reviewed in the commit:[bba01140837453f908a9dafefa89308ee9a5103e](https://github.com/wazuh/wazuh-qa/commit/bba01140837453f908a9dafefa89308ee9a5103e)

File: deps/wazuh_testing/wazuh_testing/qa_ctl/provisioning/wazuh_deployment/wazuh_local_package.py (hunk 6 above)
Comment (fernandolojano) [reply deps/wazuh_testing/wazuh_testing/qa_ctl/provisioning/wazuh_deployment/wazuh_local_package.py]: The new modification can be reviewed in the commit:[bba01140837453f908a9dafefa89308ee9a5103e](https://github.com/wazuh/wazuh-qa/commit/bba01140837453f908a9dafefa89308ee9a5103e)

File: deps/wazuh_testing/wazuh_testing/qa_ctl/provisioning/wazuh_deployment/wazuh_s3_package.py (hunk 7 above)
Comment (fernandolojano) [reply deps/wazuh_testing/wazuh_testing/qa_ctl/provisioning/wazuh_deployment/wazuh_s3_package.py]: The new modification can be reviewed in the commit:[bba01140837453f908a9dafefa89308ee9a5103e](https://github.com/wazuh/wazuh-qa/commit/bba01140837453f908a9dafefa89308ee9a5103e)

File: deps/wazuh_testing/wazuh_testing/qa_ctl/provisioning/wazuh_deployment/wazuh_s3_package.py (hunk 8 above)
Comment (fernandolojano) [reply deps/wazuh_testing/wazuh_testing/qa_ctl/provisioning/wazuh_deployment/wazuh_s3_package.py]: The new modification can be reviewed in the commit:[bba01140837453f908a9dafefa89308ee9a5103e](https://github.com/wazuh/wazuh-qa/commit/bba01140837453f908a9dafefa89308ee9a5103e)

**Pull request merged**